cmd/edgectl/aes_install.go

```diff
@@ -437,166 +428,97 @@ func (i *Installer) Perform(kcontext string) error {
-	if len(aesCrds) > 0 {
-		// AES CRDs exist so there is likely an existing installation. Try to
-		// verify the existence of an Ambassador deployment in the Ambassador
-		// namespace.
-		installedVersion, err := i.GetInstalledImageVersion()
-		if err != nil {
-			i.show.Println("Failed to look for an existing installation:", err)
-			installedVersion = ""
-			// Things will likely fail when we try to apply manifests
-		}
-		switch {
-		case i.version == installedVersion:
-			alreadyApplied = true
-			i.ShowWrapped(fmt.Sprintf("-> Found an existing installation of Ambassador Edge Stack %s.", i.version))
-		case installedVersion != "":
-			i.ShowWrapped(fmt.Sprintf("-> Found an existing installation of Ambassador Edge Stack %s.", installedVersion))
-			i.show.Println()
-			i.ShowWrapped(abortExisting)
-			i.show.Println()
-			i.ShowWrapped(seeDocs)
-			i.Report("fail_existing_aes", ScoutMeta{"installing", i.version}, ScoutMeta{"found", installedVersion})
-			return errors.Errorf("existing AES %s found when installing AES %s", installedVersion, i.version)
-		default:
-			i.ShowWrapped(abortCRDs)
-			i.show.Println()
-			i.ShowWrapped(seeDocs)
-			i.Report("fail_existing_crds")
-			return errors.New("CRDs found")
-		}
+		return err
 	}
 
-	if !alreadyApplied {
-		// Install the AES manifests
+	i.ShowWrapped(fmt.Sprintf("-> Checking latest version of Ambassador Edge Stack available..."))
 
-		i.ShowWrapped("Downloading images. (This may take a minute.)")
+	if err := chartDown.Download(); err != nil {
+		i.ShowWrapped(fmt.Sprintf("-> Failed to download Ambassador Edge Stack"), tryAgain, seeDocs)
+		i.Report("fail_download", ScoutMeta{"err", err.Error()})
+		return errors.Errorf("could not download")
+	}
 
-		if err := i.ShowKubectl("install CRDs", crdManifests, "apply", "-f", "-"); err != nil {
-			i.Report("fail_install_crds")
-			return err
-		}
+	// the AES version we have downloaded
+	downVersion := strings.Trim(chartDown.GetChart().AppVersion, "\n")
 
-		if err := i.ShowKubectl("wait for CRDs", "", "wait", "--for", "condition=established", "--timeout=90s", "crd", "-lproduct=aes"); err != nil {
-			i.Report("fail_wait_crds")
-			return err
-		}
+	chartRel, err := chartDown.GetReleaseMgr(defInstallNamespace, chartValues)
+	defer func() { _ = chartDown.Cleanup() }()
+	if err != nil {
+		i.log.Printf("when obtaining the chart manager: %s", err)
+		i.Report("fail_download", ScoutMeta{"err", err.Error()})
+		return err
+	}
 
-		if err := i.ShowKubectl("install AES", aesManifests, "apply", "-f", "-"); err != nil {
-			i.Report("fail_install_aes")
-			return err
-		}
+	if err := chartRel.Sync(i.ctx); err != nil {
+		i.log.Printf("Failed to sync release: %s", err)
+		i.Report("fail_download", ScoutMeta{"err", err.Error()})
+		return err
+	}
+
+	if !chartRel.IsInstalled() {
+		// Ambassador is not installed at all: perform the installation
+		i.SetMetadatum("AES version being installed", "aes_version", downVersion)
+		i.version = downVersion
```

Comment:
I think we should perform these steps unconditionally, like the original code did. In this version, these won't be set if the chart is already installed.

Reply:
If you mean to report the version of the Chart that is already installed, the API we are using does not provide that information. It provides the version for the Helm Chart that is currently installed though, but I'm not sure that would be useful...

Reply:
I have modified `GetInstalledImageVersion` for getting the current version of Ambassador, even if it was installed with Helm.